examples/convert_subnet.go

```diff
@@ -0,0 +1,184 @@
+//go:build convert_subnet
+// +build convert_subnet
+
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/account"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/network"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/wallet/local"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/wallet/types"
+
+	pchainTxs "github.com/ava-labs/avalanche-tooling-sdk-go/wallet/txs/p-chain"
+	avagoTxs "github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+// ConvertSubnet demonstrates converting a subnet to L1 using the wallet
+// Required environment variables:
+//   - PRIVATE_KEY: Hex-encoded private key (e.g., "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027")
+//   - SUBNET_AUTH_KEY: P-Chain bech32 address with subnet auth rights (e.g., "P-fuji1zwch24mn3sjkahds98fjd0asudjk2e4ajduu")
+//   - SUBNET_ID: The subnet ID to convert (e.g., "2DeHa7Qb6sufPkmQcFWG2uCd4pBPv9WB6dkzroiMQhd1NSRtof")
+//   - CHAIN_ID: Blockchain ID where validator manager contract is deployed (e.g., "2CA7jYxRCd5K3qS23JhPybaMsMHSsLc3vGTJxSFDvnBSNdPELz")
+//   - NODE_ID: NodeID of the bootstrap validator (e.g., "NodeID-GWPcbFJZFfZreETSoWjPimr846mXEKCtu")
+//   - BLS_PUBLIC_KEY: BLS public key for the validator (e.g., "0x80b7851ce335cee149b7cfffbf6cf0bbca3c9b25026a24056e610976d095906e833a66d5ca5c56c23a3fe50e8785a81f")
+//   - BLS_PROOF_OF_POSSESSION: BLS proof of possession (e.g., "0x89e1d6d47ff04ec0c78501a029865140e9ec12baba75a95bfc5710b3fecb8db4b6cecb5ccb1136e19f88db0539deb4420306dd60145024197b41cf89179790f20146fba398bc4d13e08540ea812207f736ca007275e4ebdb840065fdb38573de")
+//   - CHANGE_OWNER_ADDRESS: P-Chain bech32 address for remaining balance owner (e.g., "P-fuji1zwch24mn3sjkahds98fjd0asudjk2e4ajduu")
+//   - VALIDATOR_MANAGER_ADDRESS: EVM address of validator manager contract (optional, defaults to "0x0FEEDC0DE0000000000000000000000000000000")
+//   - VALIDATOR_WEIGHT: Weight for the validator (optional, defaults to 100)
+//   - VALIDATOR_BALANCE: Balance in nAVAX for the validator (optional, defaults to 1000000000)
+func ConvertSubnet() error {
+	ctx, cancel := utils.GetTimedContext(120 * time.Second)
+	defer cancel()
+
+	// Get required environment variables
+	// PRIVATE_KEY should be a hex-encoded private key string (64 characters)
+	// Example: "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
+	privateKey := os.Getenv("PRIVATE_KEY")
```

Comment:
Resurfacing Get Env vs using var in examples 
https://github.com/ava-labs/avalanche-tooling-sdk-go/pull/191#discussion_r2504562723

Reply:
would like to hear second opinions @jasonatran